dask_expr/_collection.py

```diff
@@ -93,16 +93,20 @@ def _wrap_unary_expr_op(self, op=None):
 #
 # Collection classes
 #
-
-
-class FrameBase(DaskMethodsMixin):
+from dask.typing import NewDaskCollection
+# Note: subclassing isn't required. This is just for the prototype to have a
+# check for abstractmethods but the runtime checks for duck-typing/protocol only
+class FrameBase(DaskMethodsMixin, NewDaskCollection):
     """Base class for Expr-backed Collections"""
 
     __dask_scheduler__ = staticmethod(
         named_schedulers.get("threads", named_schedulers["sync"])
     )
     __dask_optimize__ = staticmethod(lambda dsk, keys, **kwargs: dsk)
 
+    def __dask_tokenize__(self):
+        return self.expr._name
```

Comment:
This is just defined as part of the collections protocol. Not sure if it is actually required